src/main/webapp/app/shared/modal/MutationModal/AddExonForm.tsx

```diff
@@ -0,0 +1,265 @@
+import React, { useEffect, useMemo, useState } from 'react';
+import { IRootStore } from 'app/stores';
+import { componentInject } from 'app/shared/util/typed-inject';
+import { ProteinExonDTO } from 'app/shared/api/generated/curation';
+import { Col, Row } from 'reactstrap';
+import { components, OptionProps } from 'react-select';
+import CreatableSelect from 'react-select/creatable';
+import _ from 'lodash';
+import { parseAlterationName } from 'app/shared/util/utils';
+import { AsyncSaveButton } from 'app/shared/button/AsyncSaveButton';
+import { notifyError } from 'app/oncokb-commons/components/util/NotificationUtils';
+import { EXON_ALTERATION_REGEX } from 'app/config/constants/regex';
+import LoadingIndicator from 'app/oncokb-commons/components/loadingIndicator/LoadingIndicator';
+import classNames from 'classnames';
+import InfoIcon from 'app/shared/icons/InfoIcon';
+import { FaArrowLeft } from 'react-icons/fa';
+import * as styles from './styles.module.scss';
+
+export interface IAddExonMutationModalBody extends StoreProps {
+  hugoSymbol: string;
+  defaultExonAlterationName?: string;
+}
+
+type ProteinExonDropdownOption = {
+  label: string;
+  value: { exon: ProteinExonDTO; name: string } | string;
```

Comment:
Do we want use an object as the dropdown value? Couldn't a rerender potentially cause weird behavior?